File: XperiaModules/XperiaDirac/src/com/xperia/settings/dirac/BootCompletedReceiver.java
/*
 * Copyright (C) 2023 XperiaLabs Project
 * SPDX-License-Identifier: Apache-2.0
 */

package com.xperia.settings.dirac;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.util.Log;

import com.xperia.settings.dirac.DiracUtils;

public class BootCompletedReceiver extends BroadcastReceiver {

    private static final boolean DEBUG = false;
    private static final String TAG = "XperiaDIRAC";

    @Override
    public void onReceive(final Context context, Intent intent) {
        if (DEBUG) Log.d(TAG, "Received boot completed intent");
        try {
            DiracUtils.getInstance(context);
        } catch (Exception e) {
            Log.d(TAG, "DIRAC is not present in system");
        }
    }
}

File: XperiaModules/XperiaDirac/src/com/xperia/settings/dirac/BootCompletedReceiver.kt
/*
 * Copyright (C) 2023 XperiaLabs Project
 * SPDX-License-Identifier: Apache-2.0
 */

package com.xperia.settings.dirac

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.xperia.settings.dirac.DiracUtils

class BootCompletedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {

        val millis = 1 * 60 * 1000  // 1min
        try {
            Thread.sleep(millis.toLong())
            context?.let { DiracUtils(it).onBootCompleted() }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}